iothub_client/src/iothub_client_ll.c

```diff
@@ -2114,3 +2152,30 @@ IOTHUB_CLIENT_RESULT IoTHubClient_LL_UploadMultipleBlocksToBlob(IOTHUB_CLIENT_LL
 }
 
 #endif /* DONT_USE_UPLOADTOBLOB */
+
+IOTHUB_CLIENT_RESULT IoTHubClient_LL_EnableDiagnostic(IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle, int samplingPercentage)
+{
+    IOTHUB_CLIENT_LL_HANDLE_DATA* handleData = (IOTHUB_CLIENT_LL_HANDLE_DATA*)iotHubClientHandle;
+
+    if (samplingPercentage == USE_SERVER_DIAG_SETTINGS)
+    {
+        if (handleData->IoTHubTransport_Subscribe_DeviceTwin(handleData->transportHandle) != 0)
```

Comment:
what if user already subscribe device twin?

Reply:
If user already subscribe the device twin, then it will return success and without any side effects.
Related code:

![image](https://user-images.githubusercontent.com/5545529/34547676-d0344ef4-f137-11e7-9edf-a294b10c12e6.png)